app/services/force/soql/preference_service.rb

```diff
@@ -26,12 +26,18 @@ def app_preferences_for_listing(opts)
       private
 
       def buildAppPreferencesFilters(opts)
+        if opts[:accessibility]
+          accessibility_filters = opts[:accessibility].split(', ')
+          accessibility_string = accessibility_filters.length == 1 ? "'#{accessibility_filters[0]}'" : "'#{accessibility_filters[0]}', '#{accessibility_filters[1]}'"
+        end
+
         filters = ''
         filters += "and Preference_All_Name__c like '%#{opts[:preference]}' " if opts[:preference].present?
         filters += "and Application__r.Name like '%#{opts[:application_number]}%' " if opts[:application_number].present?
         filters += "and Application__r.Applicant__r.First_Name__c like '%#{opts[:first_name]}%' " if opts[:first_name].present?
         filters += "and Application__r.Applicant__r.Last_Name__c like '%#{opts[:last_name]}%' " if opts[:last_name].present?
         filters += "and Application__r.Processing_Status__c = " + (opts[:status] == 'No Status' ? 'NULL' : "'#{opts[:status]}'") if opts[:status].present?
+        filters += "and Application__r.Has_ADA_Priorities_Selected__c INCLUDES (#{accessibility_string}) " if opts[:accessibility].present?
```

Comment:
I'm not super clear on how this gives us vision OR hearing impairments, is it just some sort of SOQL magic?

Reply:
It is a query to say look for any of these that include vision impairments OR hearing impairments, so it checks against the string on each one individually

https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_querying_multiselect_picklists.htm

Reply:
Your logic thought is correct, that is the way I thought of it in SQL but yes using SOQL logic

Reply:
OK, that makes sense, sort of! can you add that the comma means or in your comment about coming up with the accessibility_string above?

Reply:
added comment and unit test, was trying to add an FE test as well but was not successful